projects/universal/src/tokens/ssr-location.ts

```diff
@@ -1,5 +1,8 @@
 import {InjectionToken} from '@angular/core';
 
+import {LocationMock} from '../classes/location-mock';
+
 export const SSR_LOCATION = new InjectionToken<Location>(
     'Location object passed from server side',
+    {factory: () => new LocationMock()},
```

Comment:
You cannot use factory in this token, unfortunately. It takes precedence over providing it because of some weird Injector hierarchy in case of SSR, I couldn't figure it out completely. Instead make them optional and provide fallback here:
https://github.com/ng-web-apis/universal/blob/master/projects/universal/src/constants/universal-user-agent.ts
https://github.com/ng-web-apis/universal/blob/master/projects/universal/src/constants/universal-location.ts

Reply:
✅